src/throttler.guard.ts

```diff
@@ -73,27 +73,44 @@ export class ThrottlerGuard implements CanActivate {
       }
     }
     const tracker = this.getTracker(req);
-    const key = this.generateKey(context, tracker);
-    const ttls = await this.storageService.getRecord(key);
-    const nearestExpiryTime = ttls.length > 0 ? Math.ceil((ttls[0] - Date.now()) / 1000) : 0;
+    const trackers = Array.isArray(tracker) ? tracker : [tracker];
+    let maxTtlsLen = 0;
+    let nearestExpiryTime = 0;
+    const keys = [];
+    const promises = trackers.map(
+      (tracker) =>
+        new Promise<void>(async (resolve) => {
+          const key = this.generateKey(context, tracker);
+          const ttls = await this.storageService.getRecord(key);
+          if (maxTtlsLen < ttls.length) {
+            maxTtlsLen = ttls.length;
+            const expiryTime = ttls.length > 0 ? Math.ceil((ttls[0] - Date.now()) / 1000) : 0;
+            nearestExpiryTime = Math.max(nearestExpiryTime, expiryTime);
+          }
+          keys.push(key);
+          resolve();
+        }),
+    );
+    await Promise.all(promises);
```

Comment:
I believe this can be simplified to:

```ts
const whenTrackersKey = trackers.map(
  async (tracker) => {
    const key = this.generateKey(context, tracker);
    const ttls = await this.storageService.getRecord(key);
    if (maxTtlsLen < ttls.length) {
      maxTtlsLen = ttls.length;
      const expiryTime = ttls.length > 0 ? Math.ceil((ttls[0] - Date.now()) / 1000) : 0;
      nearestExpiryTime = Math.max(nearestExpiryTime, expiryTime);
    }
    return key;
  }
);
const keys = await Promise.all(whenTrackersKey);
```

Reply:
Oh, so nice